pkg/log/deleg.go

```diff
@@ -25,6 +27,7 @@ import (
 type loggerPromise struct {
 	logger        *DelegatingLogger
 	childPromises []*loggerPromise
+	promisesLock  *sync.Mutex
```

Comment:
shouldn't need this to be a pointer since the containing object is already being passed around as a pointer.

Reply:
I've pushed a commit to fix this, will test within my codebase, can squash if required 😄